scripts/o3de/tests/unit_test_print_registration.py

```diff
@@ -0,0 +1,383 @@
+        pytest.param("--all-templates", pathlib.Path("D:/MinimalProject"), 1),
+        pytest.param("--all-templates", None, 1)
+    ])
+    def test_print_template_registration(self, arg_option, project_path, verbose):
+        parser = argparse.ArgumentParser()
+
+        # Register the registration script subparsers with the current argument parser
+        print_registration.add_parser_args(parser)
+        arg_list = [arg_option]
+        if project_path:
+            arg_list += ['--project-path', project_path.as_posix()]
+        if verbose:
+            arg_list += ['-' + 'v' * verbose]
+        test_args = parser.parse_args(arg_list)
+
+
+        with patch('o3de.manifest.load_o3de_manifest', side_effect=self.load_manifest_json) as load_manifest_patch, \
+                patch('o3de.manifest.get_template_json_data', side_effect=self.get_template_json_data) as get_json_patch, \
+                patch('o3de.manifest.get_project_json_data', side_effect=self.get_project_json_data) as get_project_json_patch, \
+                patch('o3de.print_registration.get_project_path', return_value=project_path) as get_project_path_patch:
+            result = print_registration._run_register_show(test_args)
+            assert result == 0
+
+
+    @pytest.mark.parametrize("arg_option, project_path, verbose", [
+        pytest.param("--restricted", None, 0),
+        pytest.param("--restricted", None, 1),
+        pytest.param("--engine-restricted", None, 0),
+        pytest.param("--engine-restricted", None, 1),
+        pytest.param("--project-restricted", pathlib.Path("D:/MinimalProject"), 0),
+        pytest.param("--project-restricted", pathlib.Path("D:/MinimalProject"), 1),
+        pytest.param("--all-restricted", pathlib.Path("D:/MinimalProject"), 0),
+        pytest.param("--all-restricted", None, 0),
+        pytest.param("--all-restricted", pathlib.Path("D:/MinimalProject"), 1),
+        pytest.param("--all-restricted", None, 1)
+    ])
+    def test_print_restricted_registration(self, arg_option, project_path, verbose):
+        parser = argparse.ArgumentParser()
+
+        # Register the registration script subparsers with the current argument parser
+        print_registration.add_parser_args(parser)
+        arg_list = [arg_option]
+        if project_path:
+            arg_list += ['--project-path', project_path.as_posix()]
+        if verbose:
+            arg_list += ['-' + 'v' * verbose]
+        test_args = parser.parse_args(arg_list)
+
+        with patch('o3de.manifest.load_o3de_manifest', side_effect=self.load_manifest_json) as load_manifest_patch, \
+                patch('o3de.manifest.get_restricted_json_data', side_effect=self.get_restricted_json_data) as get_json_patch, \
+                patch('o3de.manifest.get_project_json_data', side_effect=self.get_project_json_data) as get_project_json_patch, \
+            patch('o3de.print_registration.get_project_path', return_value=project_path) as get_project_path_patch:
```

Comment:
This line looks like it should be indented more.

Reply:
I'll update the indentation.
In this case it is cosmetic, not functional since it is part of the same `with` statement